.circleci/config.yml

```diff
@@ -0,0 +1,262 @@
+          command: |
+            echo 'export BRANCH=${CIRCLE_BRANCH//\//-}' >> "$BASH_ENV"
+            echo 'export BRANCH_IMAGE_TAG=${BRANCH}<<# parameters.squash >>-squash<</ parameters.squash >>' >> "$BASH_ENV"
+            echo 'export BRANCH_IMAGE_URL=<< parameters.image-name >>:${BRANCH_IMAGE_TAG}' >> "$BASH_ENV"
+            echo 'export SHA1_IMAGE_TAG=${CIRCLE_SHA1}<<# parameters.squash >>-squash<</ parameters.squash >>' >> "$BASH_ENV"
+            echo 'export SHA1_IMAGE_URL=<< parameters.image-name >>:${SHA1_IMAGE_TAG}' >> "$BASH_ENV"
+            echo 'export IMAGE_TAG_LIST=${SHA1_IMAGE_TAG},${BRANCH_IMAGE_TAG}' >> "$BASH_ENV"
+      # Only create the GIT_TAG related tags if we're building a tag
+      - when:
+          condition: << pipeline.git.tag >>
+          steps:
+            - run:
+                name: Create git tag related image tag vars
+                command: |
+                  echo 'export GIT_TAG_IMAGE_TAG=${CIRCLE_TAG}<<# parameters.squash >>-squash<</ parameters.squash >>' >> "$BASH_ENV"
+                  echo 'export GIT_TAG_IMAGE_URL=<< parameters.image-name >>:${GIT_TAG_IMAGE_TAG}' >> "$BASH_ENV"
+                  echo 'export IMAGE_TAG_LIST=${SHA1_IMAGE_TAG},${BRANCH_IMAGE_TAG},${GIT_TAG_IMAGE_TAG}' >> "$BASH_ENV"
+
+  save-images:
+    parameters:
+      tmp-dir:
+        type: string
+        default: "/tmp/images"
+      images:
+        type: string
+    steps:
+      - run:
+          name: Persist Docker images to tar file
+          command: |
+            mkdir -p << parameters.tmp-dir >>
+            docker save << parameters.images >> -o << parameters.tmp-dir >>/build-image
+      - persist_to_workspace:
+          name: Persist the workspace for later use
+          root: << parameters.tmp-dir >>
+          paths:
+            - build-image
+
+  load-images:
+    parameters:
+      tmp-dir:
+        type: string
+        default: "/tmp/images"
+    steps:
+      - attach_workspace:
+          name: Attach to previously created workspace
+          at: << parameters.tmp-dir >>
+      - run:
+          name: Load Docker images
+          command: docker load -i << parameters.tmp-dir >>/build-image
+
+  build-image:
+    parameters:
+      step-name:
+        type: string
+        default: "Docker build"
+      build-target:
+        type: enum
+        default: "build-image"
+        enum: ["build-image", "build-image-test"]
+      image-name:
+        type: string
+      image-tags:
+        type: string
+        default: "${CIRCLE_SHA1}"
+      squash:
+        type: boolean
+        default: false
+    steps:
+      - docker/build:
+          step-name: << parameters.step-name >>
+          image: << parameters.image-name >>
+          extra_build_args: "--build-arg NF_IMAGE_VERSION=${CIRCLE_SHA1} --build-arg NF_IMAGE_TAG=${BRANCH} --target << parameters.build-target >> <<# parameters.squash >> --squash <</ parameters.squash >>"
+          tag: << parameters.image-tags >>
+
+jobs:
```

Comment:
You may have already checked this out, but given the size of the image this produces, it could be worth bumping the executor size to see if it lowers the build time at all.